src/components/tooltip/Marker.js

```diff
@@ -714,6 +717,43 @@ class MarkerTooltipContents extends React.PureComponent<Props> {
       implementationFilter,
       zeroAt
     );
+
+    // FileIO markers can belong to different threads.
+    let threadDetail = null;
+    const data = marker.data;
+    if (data && data.type === 'FileIO' && data.threadId !== undefined) {
+      const theadId = data.threadId;
+      const occurringThreadName = threadIdToNameMap.get(theadId);
+      threadDetail = [
+        <TooltipDetail label="Recording Thread" key="recording">
+          {threadName}
+        </TooltipDetail>,
+      ];
+
+      // If we have the thread information of the occurring thread, then show.
+      // Otherwise, only show the thread ID.
+      if (occurringThreadName !== undefined) {
+        threadDetail.push(
+          <TooltipDetail
+            label="Occurring Thread"
+            key="occurring"
+          >{`${occurringThreadName} (${theadId})`}</TooltipDetail>
```

Comment:
When I tried the patch, I wasn't sure how to parse this number between parentheses. How about adding `TID: ` as prefix? `(TID: ${threadId})`.